gpflow/kernels/statics.py

```diff
@@ -52,7 +52,15 @@ class Constant(Static):
 
     def K(self, X, X2=None):
         if X2 is None:
-            shape = [tf.shape(X)[0], tf.shape(X)[0]]
+            shape = tf.concat(
+                [
+                    tf.shape(X)[:-2],
+                    tf.reshape(tf.shape(X)[-2], [1]),
+                    tf.reshape(tf.shape(X)[-2], [1]),
+                ],
+                0,
```

Comment:
nit:
```suggestion
                axis=0,
```
makes it more immediately clear what's going on